### PR TITLE
Introduce Store/Remove actions

### PR DESCRIPTION
**Pull request by jhs7jhs**
To-do:
- [x] Unit tests on Text/Action/TopLevel
- [x] Integration tests (with the SDK)

### CODE REVIEW

File: state/src/impls/top_level.rs [hunk 1]
```diff
@@ -773,6 +807,33 @@ impl TopState for TopLevelState {
...
+        Ok(())
+    }
+
+    fn remove_text(&mut self, key: &H256, sig: &Signature) -> StateResult<()> {
+        let text = self.get_text(key)?.ok_or_else(|| ParcelError::TextNotExist)?;
+        match verify_address(text.certifier(), sig, key) {
+            Ok(false) => {
+                return Err(ParcelError::TextVerificationFail("Certifier and signer are different".to_string()).into())
+            }
+            Err(err) => return Err(ParcelError::TextVerificationFail(format!("{}", err)).into()),
```
Comment (sgkim126) [inline state/src/impls/top_level.rs]:
```rust
Err(err) => return Err(ParcelError::TextVerificationFail(err.to_string()).into()),
```

File: state/src/item/text.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@ use rlp::{Decodable, DecoderError, Encodable, RlpStream, UntrustedRlp};
 use crate::CacheableItem;
 
 /// Text stored in the DB. Used by Store/Remove Action.
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, Serialize, PartialEq)]
```
Comment (sgkim126) [inline state/src/item/text.rs]: Would you define new type on the RPC module?

Comment (jhs7jhs) [reply state/src/item/text.rs]: Fixed!

File: state/src/impls/top_level.rs (hunk 1 above)
Comment (jhs7jhs) [reply state/src/impls/top_level.rs]: Fixed

File: state/src/impls/top_level.rs [hunk 3]
```diff
@@ -1158,7 +1230,7 @@ mod tests_parcel {
                 expected: 0,
                 found: 2
             }))),
-            state.apply(&parcel, &sender_public, &get_test_client())
+            state.apply(&parcel, &H256::zero(), &sender_public, &get_test_client())
```
Comment (sgkim126) [inline state/src/impls/top_level.rs]: How about using `H256::random()` for parcel_hash?

File: state/src/item/text.rs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    /// Get blake hash of the content of the text
+    pub fn content_hash(&self) -> H256 {
+        let rlp = self.content.rlp_bytes();
+        Blake::blake(rlp)
+    }
+}
+
+impl Default for Text {
```
Comment (sgkim126) [inline state/src/item/text.rs]: You can derive `Default` if the members are initialized by `Default`

File: state/src/item/text.rs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            certifier: Default::default(),
+        }
+    }
+}
+
+impl CacheableItem for Text {
+    type Address = H256;
+    /// Check if content is empty and certifier is null.
+    fn is_null(&self) -> bool {
+        self.content.is_empty() && self.certifier == Default::default()
```
Comment (sgkim126) [inline state/src/item/text.rs]:
```rust
self.content.is_empty() && self.certifier.is_zero()
```

File: spec/JSON-RPC.md [hunk 6]
```diff
@@ -99,6 +99,18 @@ A string that starts with "(NetworkID)c", and Bech32 string follows. For example
  - parameters: `number[][]`
  - amount: `U64`
 
+### Text Action
```
Comment (sgkim126) [inline spec/JSON-RPC.md]: Store

File: spec/JSON-RPC.md [hunk 7]
```diff
@@ -99,6 +99,18 @@ A string that starts with "(NetworkID)c", and Bech32 string follows. For example
...
+
+ - action: "text"
+ - content: `string`
+ - certifier: `PlatformAddress`
+ - signature: `Signature`
+
+### Invalidate Action
```
Comment (sgkim126) [inline spec/JSON-RPC.md]: Remove

File: state/src/impls/top_level.rs (hunk 3 above)
Comment (jhs7jhs) [reply state/src/impls/top_level.rs]: Fixed

**Pull request merged**